hal/ia32/arch/types.h

```diff
@@ -17,52 +17,22 @@
 #ifndef _HAL_IA32_TYPES_H_
 #define _HAL_IA32_TYPES_H_
 
-#define NULL 0
 
-#ifndef __ASSEMBLY__
-
-typedef unsigned char u8;
-typedef unsigned short u16;
-typedef unsigned int u32;
-typedef unsigned long long u64;
+#include "include/types.h"
 
-typedef u8 ld80[10]; /* Long double as bytes */
-typedef struct {
-	u16 _controlWord, controlWord;
-	u16 _statusWord, statusWord;
-	u16 _tagWord, tagWord;
-	u32 fip;
-	u32 fips;
-	u32 fdp;
-	u16 _fds, fds;
-	ld80 fpuContext[8];
-} fpu_context_t;
 
-typedef char s8;
-typedef short s16;
-typedef int s32;
-typedef long long s64;
+#ifndef __ASSEMBLY__
 
-typedef u32 addr_t;
-typedef u64 cycles_t;
 
-typedef u64 usec_t;
-/* FIXME: offs_t should be eradicated */
-typedef s64 offs_t;
-typedef offs_t off_t;
+typedef __u8 ld80[10]; /* Long double as bytes */
 
-typedef unsigned int size_t;
-typedef unsigned long long time_t;
+typedef __u64 cycles_t;
+typedef __u32 ptr_t;
 
-typedef u32 ptr_t;
+typedef unsigned long size_t;
```

Comment:
`ssize_t` for IA32 is defined to be `int` (not long) in current PR ([include/arch/ia32/types.h:29](https://github.com/phoenix-rtos/phoenix-rtos-kernel/pull/518/files#diff-3036bc66d98d2f7860b3b162b835ad5c753b76ec76be4963c9a67dd053865151R29))
Why `size_t` typedef changed to `unsigned long int` from`unsigned int` on IA32  ?

Reply:
I checked how the toolchain defines `size_t` on IA32 and in `stddef.h` which is used in userspace it's defines as `unsigned long` so I've changed it, thanks for spotting the `ssize_t`